### PR TITLE
Add Winget Releaser workflow

### PR DESCRIPTION
**Pull request by sitiom**
This action automatically generates manifests for the Winget Community Repository ([microsoft/winget-pkgs](https://github.com/microsoft/winget-pkgs)).

I have just added typioca to Winget (https://github.com/microsoft/winget-pkgs/pull/94813). This workflow would enable automatic package updates for it.

**Before merging this:**
1. Add a PAT with `public_repo` scope as a repository secret named `WINGET_TOKEN`.

example

2. Fork https://github.com/microsoft/winget-pkgs under @bloznelis. The action will use that fork for making a branch and creating a PR with the upstream [winget-pkgs](https://github.com/microsoft/winget-pkgs) repository on every release.
3. Install [Pull](https://github.com/apps/pull) on the winget-pkgs fork to ensure that it is constantly updated.

The documentation and source code of the action is available [here](https://github.com/vedantmgoyal2009/winget-releaser/).

If you want to see an example of a PR created using this action, see [microsoft/winget-pkgs/pulls (Pull request has been automatically created using 🛫 WinGet Releaser)](https://github.com/microsoft/winget-pkgs/pulls?q=is%3Apr+sort%3Aupdated-desc+Pull+request+has+been+automatically+created+using+%F0%9F%9B%AB+WinGet+Releaser).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -62,3 +62,12 @@ jobs:
...
           files: execs/**/*
+  winget:
+    needs: release
+    runs-on: windows-latest # Action can only be run on Windows
+    steps:
+      - uses: vedantmgoyal2009/winget-releaser@v1
+        with:
+          identifier: bloznelis.typioca
+          version: ${{ github.ref_name }}
+          token: ${{ secrets.WINGET_TOKEN }}
```
Comment (bloznelis) [inline .github/workflows/ci.yml:73]:
```suggestion
          token: ${{ secrets.PUBLIC_ACCESS_TOKEN }}
```

Comment (bloznelis) [reply .github/workflows/ci.yml:73]: We already have a token with `public_repo` access in this project, we use it for homebrew releases. Let's re-use it

Comment (sitiom) [reply .github/workflows/ci.yml:73]: I have overlooked this, Thanks!

**Review by bloznelis — APPROVED** [review]
Nice! Thank you for doing this. I will merge and trigger a new release. Please check if everything is ok from the winget side afterward.

**Comment by sitiom** [comment]
Winget Releaser error seems to be about this one: https://github.com/vedantmgoyal2009/winget-releaser/issues/86. Try rerunning the failed job until it succeeds.

**Comment by sitiom** [comment]
- All good now: https://github.com/microsoft/winget-pkgs/pull/95128

**Comment by sitiom** [comment]
Don't forget to sign the CLA: https://github.com/microsoft/winget-pkgs/pull/95128#issuecomment-1409908187

**Pull request merged**